app/questions.py

```
@@ -0,0 +1,340 @@
+"""
+
+from dataclasses import dataclass
+from typing import Optional
+
+TAMIL_NUMERALS = ["௦", "௧", "௨", "௩", "௪", "௫", "௬", "௭", "௮", "௯"]
+
+
+def to_tamil_numeral(n: int) -> str:
+    """Convert a non-negative integer to Tamil numerals."""
+    if not isinstance(n, int) or n < 0:
+        raise ValueError(f"Expected non-negative integer, got: {n}")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)